### DISCUSSION TITLE
Divide and run tests

### DISCUSSION THREAD

**Discussion by amitmthorat** [Banno/Gordon#128] [Q&A] [answered]
Hi All, I wanted to know if there is any way where I can spilt the tests and run them on multiple devices. 
I have 7556 instrumentation tests if I run them on single device it takes around 5-6hrs. I want to reduce this time.
Please suggest if there is any way.
Thanks!!

**Comment by joshschriever** [comment]
Yes, you can take a look at [the pooling strategies documentation](https://github.com/Banno/Gordon#pooling-strategies). Sounds like `SinglePool` is probably what you're looking for, so something like this in your Gradle buildscript:

```
import com.banno.gordon.PoolingStrategy

gordon {
    poolingStrategy.set(PoolingStrategy.SinglePool)
}
```

**Reply by amitmthorat** [reply]
Hi @joshschriever thanks for the reply. I already tried to set the pooling strategy as suggested but each test runs on all the connected devices.
Is there any other configuration I need to do?

**Reply by joshschriever** [reply]
Hmm, I just did a test with the latest Gordon version, using `SinglePool`, and verified that it splits tests up across all connected devices as expected. Do you think you could share a simple demo app with your same configuration that reproduces the issue?

**Reply by amitmthorat** [reply]
Hi @joshschriever, thanks again for checking. I'm using v1.8.7 due to some limitations. This version also has the SinglePool strategy but as I observed before it does not split the tests. Please do let me know if I need to configure anything else.

apply plugin: "com.banno.gordon"

gordon {
    PoolingStrategy.SinglePool
}

above is my app gradle code.

**Reply by joshschriever** [reply, marked as answer]
Ah, looks like you're using a groovy buildscript? Check out [the example for that](https://github.com/Banno/Gordon#groovy-buildgradle-example-for-poolingstrategy).

```
gordon {
    poolingStrategy.set(PoolingStrategy.SinglePool.INSTANCE)
}
```

**Reply by amitmthorat** [reply]
Thank you so much @joshschriever!
